internal/server/data/grant.go

```diff
@@ -19,7 +40,26 @@ func CreateGrant(db GormTxn, grant *models.Grant) error {
 	case grant.Resource == "":
 		return fmt.Errorf("resource is required")
 	}
-	return add(db, grant)
+
+	// Use a savepoint so that we can query for the duplicate grant on conflict
+	if _, err := tx.Exec("SAVEPOINT beforeCreate"); err != nil {
+		// ignore "not in a transaction" error, because outside of a transaction
+		// the db conn can continue to be used after the conflict error.
+		if !isPgErrorCode(err, pgerrcode.NoActiveSQLTransaction) {
+			return err
+		}
```

Comment:
Previously we were ignoring all errors. We have to ignore this one error code because we have plenty of tests that don't use a transaction to setup the fixture data. In the future as we add more transactions to tests maybe we'll be able to remove this.